feat: Add docs for new native features (#1212)
getsentry/sentry-docs: src/collections/_documentation/cli/dif.md

Sentry. The command will recurively scan the provided folders or ZIP archives.
Files that have already been upload are skipped automatically.

{% capture __alert_content -%
Because debug files belong to projects, you will need to specify the
organization and project you are working with. For more information about this
refer to [Working with Projects]({%- link _documentation/cli/configuration.md
-%}#sentry-cli-working-with-projects).
{%- endcapture -%}
{%- include components/alert.html
  title="Note"
  content=__alert_content
%}

A basic debug file upload can be started with:

```bash
$ sentry-cli upload-dif -o <org> -p <project> /path/to/files

> Found 2 debug information files
> Prepared debug information files for upload
> Uploaded 2 missing debug information files
> File processing complete:

     OK 1ddb3423-950a-3646-b17b-d4360e6acfc9 (MyApp; x86_64 executable)
     OK 1ddb3423-950a-3646-b17b-d4360e6acfc9 (MyApp; x86_64 debug companion)
```

### Upload Options

There are a few options you can supply to the upload command:

`--no-unwind`

: Do not scan for stack unwinding information. Specify this flag for builds with

>   disabled FPO, or when stackwalking occurs on the device. This usually excludes

Reviewer: ```suggestion
  disabled FPO, or when stack walking occurs on the device. This usually excludes
```

  executables and libraries. They might still be uploaded, if they contain debug
  information.

`--no-debug`

: Do not scan for debug information. This will usually exclude debug companion
  files. They might still be uploaded, if they contain stack unwinding
  information.

`--include-sources`

: Scans the debug files for references to source code files. If referenced files
  are available on the local file system, they are bundled and uploaded as

>   source archive. This allows to resolve source context. Only specify this

Reviewer: ```suggestion
  a source archive. This allows you to resolve source context. Only specify this
```

  command when uploading from the same machine as the build. Otherwise, use
  `difutil bundle-sources` to generate the bundle ahead of time.

`--derived-data`

> : Search for dSYMs in derived data folder. This is the default location Xcode

Reviewer: ```suggestion
: Search for dSYMs in the derived data folder. Xcode stores its build output in this default location.
```

  stores its build output in.

`--no-zips`

: By default, `sentry-cli` will open and search ZIP archives for debug files.
  This is especially useful when downloading builds from iTunes Connect or prior
  build stages in a CI environment. Use this switch to disable if your search
  paths contain large ZIP archives without debug files to speed up the search.

`--force-foreground`

> : This option forces the upload to happen in foreground. This only affects

Reviewer: ```suggestion
: This option forces the upload to happen in the foreground. This only affects
```

  uploads invoked from Xcode build steps. By default, the upload process will
  detach when started from Xcode and finish in the background. If you need to
  debug the upload process it might be useful to force the upload to run in the
  foreground.

`--info-plist`

: Overrides the search path for `Info.plist`, useful if it is located in a
  non-standard location.

`--no-reprocessing`

: This parameter prevents Sentry from triggering reprocessing right away. It can
  be useful under rare circumstances where you want to upload files in multiple
  batches and you want to ensure that Sentry does not start reprocessing before

>   some optional dsyms are uploaded. Note though that someone can still in the

Reviewer: ```suggestion
  some optional dSYMs are uploaded. Note though that someone can still in the
```

  meantime trigger reprocessing from the UI.

`--symbol-maps`

: Resolve hidden symbols in iTunes Connect builds using BCSymbolMaps. This is
  needed to symbolicate crashes if symbols were not uploaded to Apple when
  publishing the app in the AppStore.

### Symbol Maps

If you are hiding debug symbols from Apple, the debug files will not contain
many useful symbols. In that case, the sentry-cli upload will warn you that it
needs BCSymbolMaps:

```bash
$ sentry-cli upload-dif ...
> Found 34 debug information files
> Warning: Found 10 symbol files with hidden symbols (need BCSymbolMaps)
```

> In this case, you need the BCSymbolMaps that match your files. Normally, these

Reviewer: ```suggestion
In this case, you need the BCSymbolMaps that match your files. Typically, these
```

get generated by the Xcode build process. Supply the `--symbol-maps` parameter
and point it to the folder containing the symbol maps:

```bash
$ sentry-cli upload-dif --symbol-maps path/to/symbolmaps path/to/debug/symbols
```

### Breakpad Files

In contrast to native debug files, Breakpad symbols discard a lot of information
that is not required to process minidumps. Most notably, inline functions are
not declared, such that Sentry is not able to display inline frames in stack
traces.

If possible, upload native debug files such as dSYMs, PDBs or ELF files instead
of Breakpad symbols.

## ProGuard Mapping Upload

> `sentry-cli` can be used to upload ProGuard files to Sentry however in most

Reviewer: ```suggestion
`sentry-cli` can be used to upload ProGuard files to Sentry; however, in most
```

> situations you would use the [gradle

Reviewer: ```suggestion
situations, you would use the [Gradle
```

plugin](https://github.com/getsentry/sentry-java) to do that. There are some

> situations however where you would upload ProGuard files manually (for instance

Reviewer: ```suggestion
situations, however, where you would upload ProGuard files manually (for instance
```

when you only release some of the builds you are creating).

{% capture __alert_content -%}

> Because ProGuard files work on projects you will need to specify the

Reviewer: ```suggestion
You will need to specify the organization and project you are working with because ProGuard files work on projects 
```

> organization and project you are working with. For more information about this

Reviewer: ```suggestion
For more information about this
```

takes the path to one or more ProGuard mapping files and will upload them to
Sentry. If you want to associate them with an Android app you should also point
it to a processed _AndroidManifest.xml_ from a build intermediate folder.
Example:

```bash
sentry-cli upload-proguard \
    --android-manifest app/build/intermediates/manifests/full/release/AndroidManifest.xml \
    app/build/outputs/mapping/release/mapping.txt
```

Since the sentry-java SDK needs to know the UUID of the mapping file, you need
to embed it in a `sentry-debug-meta.properties` file. If you supply
`--write-properties` that is done automatically:

```bash
sentry-cli upload-proguard \
    --android-manifest app/build/intermediates/manifests/full/release/AndroidManifest.xml \
    --write-properties app/build/intermediates/assets/release/sentry-debug-meta.properties \
    app/build/outputs/mapping/release/mapping.txt
```

### Upload Options

`--no-reprocessing`

: This parameter prevents Sentry from triggering reprocessing right away. It can
  be useful under rare circumstances where you want to upload files in multiple
  batches and you want to ensure that Sentry does not start reprocessing before

>   some optional dsyms are uploaded. Note though that someone can still in the

Reviewer: ```suggestion
  some optional dSYMs are uploaded. Note though that someone can still in the
```

  meantime trigger reprocessing from the UI.

`--no-upload`

: Disables the actual upload. This runs all steps for the processing but does
  not trigger the upload (this also automatically disables reprocessing. This is
  useful if you just want to verify the mapping files and write the ProGuard

>   UUIDs into a proeprties file.

Reviewer: ```suggestion
  UUIDs into a properties file.
```
